diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/BlockStructureBracesWarnTest.kt

```diff
@@ -11,109 +12,342 @@ class BlockStructureBracesWarnTest {
-        lintMethod(BlockStructureBraces(),
+    fun `correct if expression without else`() {
+        val withBrace =
                 """
                     |fun foo() {
-                    |    if (x > 0 && x != 10) {
-                    |        bar() 
-                    |        } else {
-                    |        baz() 
-                    |        }
-                    |    
-                    |    if (x < -5) {
-                    |       baxx()
-                    |       }
-                    |    else if (x > 5) {
-                    |       rt()
-                    |       } else {
-                    |       yy() 
-                    |       }
-                    |}
-                    |
-                    |fun foo2(array: List<Int>){
-                    |
-                    |   array.forEach { model ->
-                    |       run {
-                    |           if (model % 2 == 0) {
-                    |               println(model.map { it.id })
-                    |               }
-                    |       }
+                    |   if (x > 5){
+                    |       x--
                     |   }
                     |}
                 """.trimMargin()
-        )
+        val withoutBrace =
+                """
+                    |fun foo() {
+                    |   if (x > 5)
+                    |       x--
+                    |}
+                """.trimMargin()
+        lintMethod(BlockStructureBraces(), withBrace)
+        lintMethod(BlockStructureBraces(), withoutBrace)
     }
 
     @Test
-    fun `check simple if `() {
+    fun `check correct if with else-if expression`() {
         lintMethod(BlockStructureBraces(),
                 """
                     |fun foo() {
                     |    if (x < -5) {
-                    |       bf()
+                    |       goo()
+                    |    }
+                    |    else if (x > 5) {
+                    |       hoo()
                     |    } else {
-                    |       f()
+                    |       koo() 
                     |    }
                     |}
                 """.trimMargin()
         )
     }
 
     @Test
-    fun `check wrong if `() {
+    fun `check if expression with wrong opening brace position`() {
         lintMethod(BlockStructureBraces(),
                 """
                     |fun foo() {
                     |    if (x < -5)
                     |    {
                     |       bf()
-                    |    } else {
-                    |       f()
+                    |    } else { f()
                     |    }
                     |}
                 """.trimMargin(),
-                LintError(1, 1, ruleId, "${Warnings.BRACES_BLOCK_STRUCTURE_ERROR.warnText()} open braces", false)
+                LintError(1, 1, ruleId, "${Warnings.BRACES_BLOCK_STRUCTURE_ERROR.warnText()} incorrect newline before opening brace", false),
```

Comment:
It is completely misleading that all errors point to the beginning of the file, so please change offsets in your rule and then in tests.